### ISSUE TITLE
deconv default output size not always correct

### ISSUE THREAD

**Issue by denizyuret** [denizyuret/Knet.jl#642]
See the default for PT/TF and also provide an option or a version with the output given.

**Comment by abdulbasitanees98** [comment]
Tensorflow needs an output size to be specified. However, Pytorch does not need an output size to be specified according to their definitions here:
https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/nn/conv3d_transpose
https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose3d.html
The Pytorch implementation, however, takes arguments to add zero paddings to one (specified by 'output_padding' argument) or both sides (specified by 'padding' argument) to get a desired output shape. More information about the output shape and padding is available on the PyTorch link above.
However, I think specifying the output shape and adding the 'SAME' padding as in tensorflow implementation seems more user friendly.

**Comment by abdulbasitanees98** [comment]
The PyTorch implementation also takes an optional output size parameter which calculates/ modifies the 'output_padding' parameter to match the output size specified by the user.
